# HG changeset patch
# User Torsten <mttl@mailbox.org>
# Date 1530973787 -7200
#      Sat Jul 07 16:29:47 2018 +0200
# Node ID 2d588bad7d9e4fc6aefa249d997b2c33a0a60237
# Parent  bb779fc2d0dbafd2123d74aa4f8ee3f4b00ad71a
do not set breakpoint if file is unmodified and still unnamed (bug #54231)

* file-editor-tab.cc (handle_request_remove_breakpoint): also check for a
  valid file name before setting a breakpoint via editor

diff --git a/libgui/src/m-editor/file-editor-tab.cc b/libgui/src/m-editor/file-editor-tab.cc
--- a/libgui/src/m-editor/file-editor-tab.cc
+++ b/libgui/src/m-editor/file-editor-tab.cc
@@ -514,17 +514,17 @@ namespace octave
           }
         else
           {
             if (markers_mask & ((1 << marker::breakpoint)
                                 | (1 << marker::cond_break)))
               handle_request_remove_breakpoint (editor_linenr + 1);
             else
               {
-                if (unchanged_or_saved ())
+                if (unchanged_or_saved () && valid_file_name ())
                   handle_request_add_breakpoint (editor_linenr + 1, "");
               }
           }
       }
   }
 
 
   void file_editor_tab::update_lexer (void)
